connect/transforms/src/main/java/org/apache/kafka/connect/transforms/DropHeaders.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.transforms;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.transforms.util.NonEmptyListValidator;
+import org.apache.kafka.connect.transforms.util.SimpleConfig;
+
+import java.util.List;
+import java.util.Map;
+
+public class DropHeaders<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    public static final String OVERVIEW_DOC =
+        "Drop header(s) specified in a comma-separated list containing the list of the headers(s) name(s) to remove.";
+
+    private static final String HEADERS_CONFIG = "names";
+
+    public static final ConfigDef CONFIG_DEF = new ConfigDef()
+        .define(HEADERS_CONFIG, ConfigDef.Type.LIST, ConfigDef.NO_DEFAULT_VALUE,
+            new NonEmptyListValidator(), ConfigDef.Importance.MEDIUM,
+            "Header name(s) to remove.");
+
+    private List<String> headers;
+
+    @Override
+    public void configure(Map<String, ?> props) {
+        final SimpleConfig config = new SimpleConfig(CONFIG_DEF, props);
+        headers = config.getList(HEADERS_CONFIG);
+    }
+
+    @Override
+    public R apply(R record) {
+        if (record.headers().isEmpty()) {
+            throw new DataException("The message does not contain headers.");
```

Comment:
The goal of this SMT is to remove from records all headers whose names are specified in the SMT's configuration. So we should be tolerant of a record not having any headers or even not having all of the named headers.

Secondly, we never want to modify the record being passed in; we always have to create a new record. So, it's probably better to make `headers` be a `Set<String>` (noticeably more efficient) and then do something like:
```
if (record == null) {
    return null;
}
Headers recordHeaders = record.headers();
Headers newHeaders = new ConnectHeaders();
Iterator<Header> iterator = recordHeaders.iterator();
while (iterator.hasNext()) {
    Header header = iterator.next();
    if (!headers.contains(header.key())) {
        recordHeaders.add(header);
    }
}
return record.newRecord(record.topic(), record.kafkaPartition(), record.keySchema(), record.key(), record.valueSchema(), updatedValue, record.timestamp(), newHeaders);
```